tests/test_ext_autodoc_configs.py

```diff
@@ -485,64 +485,124 @@ def test_autodoc_typehints_signature(app):
     options = {"members": None,
                "undoc-members": True}
     actual = do_autodoc(app, 'module', 'target.typehints', options)
-    assert list(actual) == [
-        '',
-        '.. py:module:: target.typehints',
-        '',
-        '',
-        '.. py:class:: Math(s: str, o: object = None)',
-        '   :module: target.typehints',
-        '',
-        '',
-        '   .. py:method:: Math.decr(a: int, b: int = 1) -> int',
-        '      :module: target.typehints',
-        '',
-        '',
-        '   .. py:method:: Math.horse(a: str, b: int) -> None',
-        '      :module: target.typehints',
-        '',
-        '',
-        '   .. py:method:: Math.incr(a: int, b: int = 1) -> int',
-        '      :module: target.typehints',
-        '',
-        '',
-        '   .. py:method:: Math.nothing() -> None',
-        '      :module: target.typehints',
-        '',
-        '',
-        '.. py:class:: NewAnnotation(i: int)',
-        '   :module: target.typehints',
-        '',
-        '',
-        '.. py:class:: NewComment(i: int)',
-        '   :module: target.typehints',
-        '',
-        '',
-        '.. py:class:: SignatureFromMetaclass(a: int)',
-        '   :module: target.typehints',
-        '',
-        '',
-        '.. py:function:: complex_func(arg1: str, arg2: List[int], arg3: Tuple[int, '
-        'Union[str, Unknown]] = None, *args: str, **kwargs: str) -> None',
-        '   :module: target.typehints',
-        '',
-        '',
-        '.. py:function:: decr(a: int, b: int = 1) -> int',
-        '   :module: target.typehints',
-        '',
-        '',
-        '.. py:function:: incr(a: int, b: int = 1) -> int',
-        '   :module: target.typehints',
-        '',
-        '',
-        '.. py:function:: missing_attr(c, a: str, b: Optional[str] = None) -> str',
-        '   :module: target.typehints',
-        '',
-        '',
-        '.. py:function:: tuple_args(x: Tuple[int, Union[int, str]]) -> Tuple[int, int]',
-        '   :module: target.typehints',
-        '',
-    ]
+    if sys.version_info < (3, 7):
+        assert list(actual) == [
+            '',
+            '.. py:module:: target.typehints',
+            '',
+            '',
+            '.. py:class:: Math(s: str, o: object = None)',
```

Comment:
Sadly, py35 and py36 does not support `Optional[object]`. I don't know why we use `object` type annotation for this case. So it would be better to change the example itself.
```
$ python3.5
Python 3.5.10 (default, Oct 24 2020, 15:43:16)
[GCC 4.2.1 Compatible Apple LLVM 11.0.3 (clang-1103.0.32.59)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import Optional
>>> Optional[int]
typing.Union[int, NoneType]
>>> Optional[object]
<class 'object'>
```